From 932d07fa40bf0ab898a2247b9a0581d7a444f620 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:14:20 -0400
Subject: [ethernet] be2net: use be_max_vfs() macro to access max-vfs

Message-id: <1410192881-30702-58-git-send-email-ivecera@redhat.com>
Patchwork-id: 92464
O-Subject: [RHEL7.1 PATCH 57/78] be2net: use be_max_vfs() macro to access max-vfs
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit d3518e215af9545eeac1046eec84cde525bae2a5
Author: Vasundhara Volam <vasundhara.volam@emulex.com>
Date:   Thu Jul 17 16:20:29 2014 +0530

    be2net: use be_max_vfs() macro to access max-vfs

    max-vfs value must be accessed via the macro be_max_vfs(adapter).
    The earlier patch "create optimal number of queues on SR-IOV config"
    by mistake, did not use this macro. This patch fixes it.

    fixes: bec84e6b ("be2net: create optimal number of queues on SR-IOV config")
    Signed-off-by: Vasundhara Volam <vasundhara.volam@emulex.com>
    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index 4213c70..fcb7dc1 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -3308,7 +3308,7 @@ static void BEx_get_resources(struct be_adapter *adapter,
  res->max_rx_qs = res->max_rss_qs + 1;
 
  if (be_physfn(adapter))
-  res->max_evt_qs = (res->max_vfs > 0) ?
+  res->max_evt_qs = (be_max_vfs(adapter) > 0) ?
      BE3_SRIOV_MAX_EVT_QS : BE3_MAX_EVT_QS;
  else
   res->max_evt_qs = 1;
-- 
1.7.1